streamparse/storm/bolt.py

```diff
@@ -446,32 +403,44 @@ def _batch_entry_run(self):
                 # later batch raises an exception
                 self._batches[key] = []
             self._batches = defaultdict(list)
+            self._tick_counter = 0
 
-    def _batch_entry(self):
-        """Entry point for the batcher thread."""
-        try:
-            while True:
-                self._batch_entry_run()
-        except Exception as e:
-            log_msg = ("Exception in {}.run() while processing tuple batch "
-                       "{!r}.".format(self.__class__.__name__,
-                                      self._current_tups))
-            log.error(log_msg, exc_info=True)
-            self.raise_exception(e, self._current_tups)
+    def process(self, tup):
```

Comment:
I tend to be more of a fan of doing nothing.  The responses I picture in my head are:

"Nothing happened, I should look at the docs to figure out why"

and 

"This is broken. Looks like it does X differently. I'll adapt my code to handle that."

Then again, this feels subjective to me, so do what you think is best. I see the merit in both approaches.